### ISSUE TITLE
Compile error for overflow literal is wrong

### ISSUE THREAD

**Issue by ytoml** [vlang/v#16705]
### Describe the bug

V compiler raises error when we use too large literal, but its boundary (where compiler decides raising error or not) seems wrong.

### Expected Behavior

Compiler raises error iff literal is larger than 2147483647 or smaller than -2147483648.

### Current Behavior

These are examples of literals not properly handled.
```v
fn cannot_compile() {
    println('${0b111_1111_1111_1111_1111}')  // error: integer literal 0b1111111111111111111 overflows int
}

fn can_compile() {
    println('${2147483648}')                    // prints -2147483648
    println('${0xFF_FF_FF_FF_FF_FF_FF_FF}')     // prints -1
    println('${0o377777_77777}')                // prints -1
    println('${-2147483649}')                   // prints 2147483647
}
```

### Reproduction Steps

See current behavior section.

### V version

V 0.3.2 9f1239c

### Environment details (OS name and version, etc.)

OS: macos, macOS, 12.5, 21G72
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 14.0.0 (clang-1400.0.29.202)

**Comment by spytheman** [comment]
Fixed in a8f6f9e .

**Comment by ytoml** [comment]
Thank you!
